### PR TITLE
[SERV-540/541] Update and Delete Institutions

### PR DESCRIPTION
**Pull request by DRickard**
* Added `updateInstitution` and `removeInstitution` methods
* Added tests for new methods
* Added utility method for making random `Institution` objects for use in tests

### CODE REVIEW

**Review by ksclarke — DISMISSED** [review]

**Review by markmatney — DISMISSED** [review]

File: src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java [hunk 1]
```diff
@@ -177,4 +166,58 @@ public final void testListInstitution(final Vertx aVertx, final VertxTestContext
...
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    public final void testDeleteInstitution(final Vertx aVertx, final VertxTestContext aContext)
+            throws AddressException, MalformedURLException, NumberParseException {
+        final Institution toDelete = TestUtils.getRandomInstitution();
+        myScheduleStoreProxy.addInstitution(toDelete).onSuccess(newID -> {
+            myScheduleStoreProxy.removeInstitution(newID.intValue()).onSuccess(result -> {
+                myScheduleStoreProxy.getInstitution(newID.intValue()).onFailure(details -> {
```
[the remark this answers is not in the document]
Comment (DRickard) [reply src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java:181]: Sounds reasonable to me. Do that in this ticket, or make a new ticket for it?

[the remark this answers is not in the document]
Comment (markmatney) [reply src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java:181]: I'm fine with whatever

File: pom.xml [hunk 2]
```diff
@@ -241,6 +241,13 @@
       <artifactId>vertx-junit5</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.jeasy</groupId>
+      <artifactId>easy-random-randomizers</artifactId>
+      <version>5.0.0</version>
```
Comment (markmatney) [inline pom.xml:247]: Like the other deps, the version number should be stored in a property that is then referenced here.

File: src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java [hunk 3]
```diff
@@ -177,4 +166,58 @@ public final void testListInstitution(final Vertx aVertx, final VertxTestContext
...
+                    final ServiceException error = (ServiceException) details;
+
+                    aContext.verify(() -> {
+                        assertEquals(Error.NOT_FOUND.ordinal(), error.failureCode());
+                        assertTrue(error.getMessage().contains(String.valueOf(newID)));
+                    }).completeNow();
+                }).onSuccess(select -> {
+                    aContext.failNow("delete failed");
+                });
+            });
```
Comment (markmatney) [inline src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java:191]: the call to `removeInstitution` should probably have a failure handler too

File: src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java [hunk 4]
```diff
@@ -177,4 +166,58 @@ public final void testListInstitution(final Vertx aVertx, final VertxTestContext
...
+                myScheduleStoreProxy.getInstitution(instID).onSuccess(updated -> {
+                    aContext.verify(() -> {
+                        assertTrue(updated.getName().equals(original.getName()));
+                        assertTrue(updated.getDescription().equals(modified.getDescription()));
+                    }).completeNow();
+                    aContext.completeNow();
+                });
+                aContext.completeNow();
+            });
+            aContext.completeNow();
```
Comment (markmatney) [inline src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java:219]: These three calls to `aContext.completeNow()` need to be removed. The test should only pass if the assertions on L212 and L213 succeed; that's handled by the chained call to `completeNow` (after `verify`) on L214. And the calls to `updateInstitution` and `getInstitution` need failure handlers.

**Review by markmatney — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (DRickard) [reply pom.xml]: Switched to version property def

File: src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java (hunk 3 above)
Comment (DRickard) [reply src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java]: Added fail handler

File: src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java (hunk 4 above)
Comment (DRickard) [reply src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java]: Added fail handler, removed wrongful completes

**Review by markmatney — APPROVED** [review]

**Review by ksclarke — APPROVED** [review]

**Pull request merged**